Bump base Alpine version and Java version

chirauki opened the pull request:
Bumps the base Alpine version to resolve some CVEs. Also bumps the minor Java version.

chirauki commented:
cc/ @llinder @jcchavezs

shakuzen commented on Dockerfile:
```diff
@@ -6,14 +6,14 @@
 # When updating, also update the README
 #  * Use current version from https://pkgs.alpinelinux.org/packages?name=openjdk15
-ARG java_version=15.0.5_p3
+ARG java_version=15.0.6_p5-r0
```
Not something that needs to block merging this pull request, but I'd like to see us get onto Java 17 since 15 should be EOL by now. Unfortunately, I don't have the time to do any thorough testing that everything works fine with Java 17, and it would be better for a user to do that in a more real (test) environment, anyway. Just some food for thought.

llinder replied:
I don't think we want `-r0` here. From what I recall it should just be `15.0.6_p5`.

shakuzen reviewed (approved):
LGTM

llinder reviewed (commented):
Looks OK, just a small question about including the `-r0` in one spot. From what I recall it was important not to include that.

chirauki commented:
The readme clearly says to strip it out when calling the build script. From the scripts, I think that ends up as the `java_version` arg in the docker build, so the default in the Dockerfile does not apply. I have removed the suffix in the docker file and added a note about it for anyone that runs into this in the future.
